### DISCUSSION TITLE
Error while installing pg_partman for postgre-12

### DISCUSSION THREAD

**Discussion by jeetendra-fareye** [pgpartman/pg_partman#550] [General]
Getting this error while installing pg_partman for postgres-12

`/usr/bin/clang-10 -Wno-ignored-attributes -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument -O2  -I. -I./ -I/usr/include/postgresql/12/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5 -flto=thin -emit-llvm -c -o src/pg_partman_bgw.bc src/pg_partman_bgw.c
make: /usr/bin/clang-10: No such file or directory
make: *** [/usr/lib/postgresql/12/lib/pgxs/src/makefiles/../../src/Makefile.global:1063: src/pg_partman_bgw.bc] Error 127
`
I have already installed clang-14 
please help with this error

**Comment by keithf4** [comment]
Do you have the postgresql source files available?

**Comment by Nadavrbn** [comment]
In case anyone else is encountering this error like me, I had a dockerfile based on postgres:16-alpine, that installed clang15 and llvm15.
Like so: 
```
FROM postgres:16-alpine

ENV PG_PARTMAN_VERSION v5.0.1

RUN apk update

RUN apk add --no-cache \
        ca-certificates \
        openssl \
        tar \
        autoconf \
        automake \
        g++ \
        clang15 \
        llvm15 \
        libtool \
        libxml2-dev \
        make \
        perl

# Install pg_partman
RUN wget -O pg_partman.tar.gz "https://github.com/pgpartman/pg_partman/archive/$PG_PARTMAN_VERSION.tar.gz" \
    # Create a folder to put the src files in
    && mkdir -p /usr/src/pg_partman \
    # Extract the src files
    && tar \
        --extract \
        --file pg_partman.tar.gz \
        --directory /usr/src/pg_partman \
        --strip-components 1 \
    # Delete src file tar
    && rm pg_partman.tar.gz \
    # Move to src file folder
    && cd /usr/src/pg_partman \
    # Build the extension
    && make \
    # Install the extension
    && make install \
    # Delete the src files for pg_partman
    && rm -rf /usr/src/pg_partman
```


```
#7 0.716 make: clang-19: No such file or directory
#7 0.716 make: *** [/usr/local/lib/postgresql/pgxs/src/makefiles/../../src/Makefile.global:1093: src/pg_partman_bgw.bc] Error 127
```

It suddenly stopped working, and the issue was that the maintainer of postgres:alpine bumped the clang and llvm versions to 19, which broke my installation setup. I had to bump the versions in my dockerfile as well.
https://github.com/docker-library/postgres/commit/17818f21dca10ccf02711476e138c219bd31b456
